Add PTZ Camera recommendations, explanations and docs for 2-way audio

janishutz opened the pull request:
## Proposed change
Added additional notes on camera support for auto-tracking and a general rule-of-thumb for how to identify if a camera is likely to work with Frigate before buying. 

This is based on about 12h of research and reading parts of the ONVIF spec and validation methods.
(Edited page: https://deploy-preview-19740--frigate-docs.netlify.app/configuration/cameras)

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code
- [x] Documentation Update


## Checklist
(no code changes happened)

- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] UI changes including text have used i18n keys and have been added to the `en` locale.
- [x] The code has been formatted using Ruff (`ruff format frigate`)

Commit:
Add PTZ Camera recommendations, explanations and docs for 2-way audio

Commit:
Fix typos

hawkeye217 commented on docs/docs/configuration/cameras.md:
```diff
@@ -89,7 +89,7 @@ An ONVIF-capable camera that supports relative movement within the field of view
 ## ONVIF PTZ camera recommendations
 
-This list of working and non-working PTZ cameras is based on user feedback.
+This list of working and non-working PTZ cameras is based on user feedback. Generally, if on the FeatureList for the camera found on the [ONVIF Conformant Products Database](https://www.onvif.org/conformant-products/), `PTZRelative`, `PTZRelativePanTilt` and / or `PTZRelativeZoom`, as well as `PTZAuxiliary` are listed, the camera is likely to work and it is guaranteed not to work if the features  ...
```
I think I'd like to avoid explicit language like "likely to work". In developing autotracking and supporting users of it over the last several years, I've found that there is sometimes no rhyme or reason why firmware devs don't implement necessary features and yet still say they "support ONVIF". Some cameras even advertise all of the appropriate features and then don't respond to any commands.

I'd prefer language like this:

> The ONVIF Conformant Products Database can provide a starting point: if the FeatureList for a camera includes PTZRelative, PTZRelativePanTilt and/or PTZRelativeZoom, along with PTZAuxiliary, the camera may support the functionality required for PTZ control. However, support is inconsistent—some cameras list these features but fail to respond to commands. If the FeatureList does not include these PTZ features, the camera will not work. If no entry exists for the camera, it is generally best to avoid purchasing it unless it is explicitly mentioned as working below.

Commit:
Change lingo as suggested

Commit:
Add issue template for camera compatibility report and ask user to fill it out in docs

janishutz commented:
Sorry for the delay in updating, I updated it and added some more findings. I will be confirming if the Dahua P5D-5F-PV works, support said that the features are supported, so I will find out, but I will open a separate PR for an update to the list. 

I have also added an issue template so users can more easily report camera compatibility with some features.

NickM-27 commented on docs/docs/configuration/cameras.md:
````diff
@@ -137,3 +141,6 @@ camera_groups:
 ```
+
+## Two-Way Audio
+See the [here](/configuration/live/#two-way-talk) for a guide on how to set up Two way audio / Two way talk and some starting points to determine if your camera supports the required features.
````
I think we could refactor this to have a single sentence linking there, and then mentioned the FeatureList with Onvif Profile T

janishutz replied:
I personally feel like there should simply be a single page (likely extra configuration or camera), where this explained and have the other places where it would be relevant also link to that page.

NickM-27 replied:
It's mostly relevant in the live view page because two way audio is done in the live view and has related configuration which is required for live view.

NickM-27 commented on docs/docs/configuration/cameras.md:
```diff
@@ -89,7 +89,9 @@ An ONVIF-capable camera that supports relative movement within the field of view
 ## ONVIF PTZ camera recommendations
 
-This list of working and non-working PTZ cameras is based on user feedback.
+This list of working and non-working PTZ cameras is based on user feedback. It is thus important that you [report camera compatibility](https://github.com/blakeblackshear/frigate/issues) if you manage or do not manage to get a camera not listed here working.
```
Yes, we definitely do not want an issue template for this:

1. Users have had no problem creating PRs to update the docs, the issue template would only be relevant if we wanted this information and didn't think users could provide it themselves via a docs PR
2. Yes, we are not trying to maintain an exhaustive or even full resource for users to check for cameras. It is mostly trying to serve as a general guide on what brands support what features. Honestly at this point reolink can likely be consolidated to a single row

janishutz committed:
Update docs as suggested

Co-authored-by: Josh Hawkins <32435876+hawkeye217@users.noreply.github.com>

Commit:
Remove issue template, tweak call to action

hawkeye217 commented on docs/docs/configuration/cameras.md:
```diff
@@ -89,7 +89,9 @@ An ONVIF-capable camera that supports relative movement within the field of view
 ## ONVIF PTZ camera recommendations
 
-This list of working and non-working PTZ cameras is based on user feedback.
+This list of working and non-working PTZ cameras is based on user feedback. It is thus important that you report your camera working or not working by opening an issue or by creating a pull request. 
```
```suggestion
This list of working and non-working PTZ cameras is based on user feedback. If you'd like to report specific quirks or issues with a manufacturer or camera that would be helpful for other users, open a pull request to add to this list.
```

The issues queue on this repo is for feature requests, and I think this wording should not feel as "urgent" to get users to report their specific camera because, as noted earlier, we're not looking for this list to be exhaustive.

hawkeye217 commented on docs/docs/configuration/live.md:
```diff
@@ -176,6 +176,8 @@ For devices that support two way talk, Frigate can be configured to use the feat
 
 To use the Reolink Doorbell with two way talk, you should use the [recommended Reolink configuration](/configuration/camera_specific#reolink-doorbell)
 
+A starting point to check compatibility for your camera, check out the list of cameras supported for two-way talk on the [go2rtc repository](https://github.com/AlexxIT/go2rtc?tab=readme-ov-file#two-way-audio). For cameras in the category `ONVIF Profile T`, you can use the [ONVIF Conformant Products Database](https://www.onvif.org/conformant-products/)'s FeatureList to check for the presence of `A ...
```
```suggestion
As a starting point to check compatibility for your camera, view the list of cameras supported for two-way talk on the [go2rtc repository](https://github.com/AlexxIT/go2rtc?tab=readme-ov-file#two-way-audio). For cameras in the category `ONVIF Profile T`, you can use the [ONVIF Conformant Products Database](https://www.onvif.org/conformant-products/)'s FeatureList to check for the presence of `AudioOutput`. A camera that supports `ONVIF Profile T` *usually* supports this, but due to inconsistent support, a camera that explicitly lists this feature may still not work. If no entry for your camera exists on the database, it is recommended not to buy it or to consult with the manufacturer's support on the feature availability.
```

janishutz committed:
Add suggested tweaks

Co-authored-by: Josh Hawkins <32435876+hawkeye217@users.noreply.github.com>

janishutz committed:
Lingo update as suggested

Co-authored-by: Josh Hawkins <32435876+hawkeye217@users.noreply.github.com>

Commit:
Shorter link to two-way audio page
